packages/block-library/src/rss/style.scss

```diff
@@ -41,3 +41,10 @@ ul.wp-block-rss { // The ul is needed for specificity to override the reset styl
 	display: block;
 	font-size: 0.8125em;
 }
+.wp-block-rss {
+	box-sizing: border-box;
+	.wp-block-rss {
```

Comment:
Same here, a comment might help future contributors to know what this does.

Reply:
I believe these styles are to prevent the double-up of block support styles within the editor when server-side rendering, hence the nested `.wp-block-rss` classes.

Given they are editor-only styles, they shouldn't be added to the styles sent to the frontend, instead `editor.scss` might be a better home for them.

These styles are missing the padding reset and the border reset might be too simplistic when Global Styles are taken into account. The tag cloud block went through these same edge cases so it's a good example to follow for the [style resets](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/tag-cloud/editor.scss?plain=1#L1-L11).

Reply:
My memory failed me during my review but finally kicked into gear. There's a slightly cleaner way to wrangle resetting all the various block support styles when skipping block supports in the server-side rendering. It should also be more forward compatible for future block supports.

It was first proposed for the Tag Cloud block that runs into similar issues.

```scss
.wp-block-rss .wp-block-rss {
	all: inherit; /* This addresses borders, background colors etc. */
	margin: 0;
	padding: 0;
}
```

Reply:
- Moved the nested style in editor.scss as per the discussion, since it is editor only style.

- One scenario, while we adding `all: inherit;` list style get inherited and bullets will display in the editor part.

  So in that case, just updated the style as follows:
  
```
  .wp-block-rss .wp-block-rss {
	  margin: 0;
	  padding: 0;
	  border: 0;
	  border-radius: inherit;
	  background: inherit;
  }
```

Please share your thought on this @aaronrobertshaw

Reply:
> Please share your thought on this @aaronrobertshaw

I think we should still use the `all: inherit;` rule. 

The issue with the `list-style: none` rule not being applied is just that the main style adding it targets the `ul` element. When the block is server-side rendered, it's wrapper becomes a `div`.

The fix is as simple as moving the existing `list-style: none;` rule to the new style you've added using the straightforward `0-1-0` specificity `.wp-block-rss` selector. Along with that move, we can also clean up the unneeded `&.is-grid` style's rule for list style as well.

The `reset.scss` styles are scoped with `:where(.editor-styles-wrapper)` nowadays so the block doesn't need the bumped specificity. Once this PR is updated though we'll need to double check a non-iframed editor and ensure everything still works as expected.

The benefit of the `all: inherit` rule is that it is more robust in the face of changes than arbitrarily adding all possible resets. As the list of CSS properties that need to be reset grows, it also means more CSS. The `all: inherit;` approach keeps that to a minimum.